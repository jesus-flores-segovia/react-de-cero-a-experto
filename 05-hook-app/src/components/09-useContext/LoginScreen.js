import React, { useContext } from 'react';
import { useForm } from '../../hooks/useForm';
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    const [values, handleInputChange, reset] = useForm({
        email: "",
        password: "",
    });

    const {email, password} = values;

    const {user, setUser} = useContext(UserContext);

    const handleSubmit = (e) => {
        e.preventDefault();
        setUser(values);
        reset();
        console.log(values);
    };

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr/>
            <form onSubmit={handleSubmit}>
                <div className="mb-3">
                    <label htmlFor="inputEmail" className="form-label">Email address</label>
                    <input name="email" type="email" className="form-control" id="inputEmail" aria-describedby="emailHelp" value={email} onChange={handleInputChange}/>
                </div>
                <div className="mb-3">
                    <label htmlFor="inputPassword" className="form-label">Password</label>
                    <input name="password" type="password" className="form-control" id="inputPassword" value={password} onChange={handleInputChange}/>
                </div>
                <button type="submit" className="btn btn-primary">Login</button>
            </form>
        </div>
    )
}
