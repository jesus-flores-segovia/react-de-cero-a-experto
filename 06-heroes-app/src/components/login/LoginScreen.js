import React, { useContext } from 'react';
import { login } from '../../actions/authActions';
import { AuthContext } from '../../auth/AuthContext';

export const LoginScreen = ({history}) => {

    const {dispatch} = useContext(AuthContext);

    const handleLogin = () => {
        dispatch(login("Jes√∫s"));
        history.replace("/");
    }

    return (
        <div className="container mt-5">
            <h1>Login</h1>
            <hr/>
            <button 
                className="btn btn-primary"
                onClick={handleLogin}>
                Login
            </button>
        </div>
    )
}
