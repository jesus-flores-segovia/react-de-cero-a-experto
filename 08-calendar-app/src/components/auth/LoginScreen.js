import React from "react";
import { useDispatch } from "react-redux";
import Swal from "sweetalert2";
import { startLogin, startRegister } from "../../actions/auth";
import { isValidEmail } from "../../helpers/validator";
import { useForm } from "../../hooks/useForm";
import "./login.css";

export const LoginScreen = () => {
  const dispatch = useDispatch();

  const [formLoginValues, handleLoginInputChange] = useForm({
    loginEmail: "",
    loginPassword: "",
  });

  const { loginEmail, loginPassword } = formLoginValues;

  const handleLogin = (e) => {
    e.preventDefault();

    if (loginEmail.trim().length === 0) {
      Swal.fire("Cannot login", "Email is mandatory", "error");
      return;
    } else if (!isValidEmail(loginEmail)) {
      Swal.fire("Cannot login", "Email is not valid", "error");
      return;
    }

    if (!loginPassword.length >= 6) {
      Swal.fire(
        "Cannot login",
        "Password must be at least of 6 characters",
        "error"
      );
    }

    dispatch(startLogin(loginEmail, loginPassword));
  };

  const [formRegisterValues, handleRegisterInputChange] = useForm({
    registerName: "",
    registerEmail: "",
    registerPassword: "",
    registerPassword2: "",
  });

  const { registerName, registerEmail, registerPassword, registerPassword2 } =
    formRegisterValues;

  const handleRegister = (e) => {
    e.preventDefault();

    if (registerName.trim().length === 0) {
      Swal.fire("Cannot register", "Name is mandatory", "error");
      return;
    }

    if (registerEmail.trim().length === 0) {
      Swal.fire("Cannot register", "Email is mandatory", "error");
      return;
    } else if (!isValidEmail(registerEmail)) {
      Swal.fire("Cannot register", "Email is not valid", "error");
      return;
    }

    if (registerPassword.length < 6) {
      Swal.fire(
        "Cannot register",
        "Password must be at least of 6 characters",
        "error"
      );
      return;
    }

    if (registerPassword !== registerPassword2) {
      Swal.fire("Cannot register", "Password fields must match", "error");
    }

    dispatch(startRegister(registerName, registerEmail, registerPassword));
  };

  return (
    <div className="container login-container">
      <div className="row">
        <div className="col-md-6 login-form-1">
          <h3>Login</h3>
          <form onSubmit={handleLogin} autoComplete="off">
            <div className="form-group">
              <input
                type="text"
                className="form-control"
                placeholder="Email"
                name="loginEmail"
                value={loginEmail}
                onChange={handleLoginInputChange}
              />
            </div>
            <div className="form-group">
              <input
                type="password"
                className="form-control"
                placeholder="Password"
                name="loginPassword"
                value={loginPassword}
                onChange={handleLoginInputChange}
              />
            </div>
            <div className="form-group">
              <input type="submit" className="btnSubmit" value="Login" />
            </div>
          </form>
        </div>

        <div className="col-md-6 login-form-2">
          <h3>Register</h3>
          <form onSubmit={handleRegister}>
            <div className="form-group">
              <input
                type="text"
                className="form-control"
                placeholder="Name"
                name="registerName"
                value={registerName}
                onChange={handleRegisterInputChange}
              />
            </div>
            <div className="form-group">
              <input
                type="email"
                className="form-control"
                placeholder="Email"
                name="registerEmail"
                value={registerEmail}
                onChange={handleRegisterInputChange}
              />
            </div>
            <div className="form-group">
              <input
                type="password"
                className="form-control"
                placeholder="Password"
                name="registerPassword"
                value={registerPassword}
                onChange={handleRegisterInputChange}
              />
            </div>

            <div className="form-group">
              <input
                type="password"
                className="form-control"
                placeholder="Repeat the password"
                name="registerPassword2"
                value={registerPassword2}
                onChange={handleRegisterInputChange}
              />
            </div>

            <div className="form-group">
              <input
                type="submit"
                className="btnSubmit"
                value="Create account"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};
