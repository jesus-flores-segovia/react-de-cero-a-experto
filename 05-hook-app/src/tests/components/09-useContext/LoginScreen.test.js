import * as React from 'react';
import { mount } from "enzyme";
import { UserContext } from "../../../components/09-useContext/UserContext";
import { LoginScreen } from "../../../components/09-useContext/LoginScreen";

const setUser = jest.fn();

const user = {
    email: "floressegoviajesus@gmail.com",
    password: "12345"
};

describe("Tests inside file 'LoginScreen.test.js'", () => {

    const wrapper = mount(
        <UserContext.Provider value={{setUser}}>
            <LoginScreen/>
        </UserContext.Provider>
    );

    test("'LoginScreen' component must show correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("'LoginScreen' component must fire the onClick event and set the user information correctly", () => {

        const inputEmail = wrapper.find("#inputEmail");
        const inputPassword = wrapper.find("#inputPassword");
        const form = wrapper.find("form");

        inputEmail.simulate("change", {
            target: {
                value: user.email,
                name: "email"
            }
        });

        inputPassword.simulate("change", {
            target: {
                value: user.password,
                name: "password"
            }
        });

        form.simulate("submit");
        expect(setUser).toHaveBeenCalledTimes(1);
        expect(setUser).toBeCalledWith(user);
    });
});