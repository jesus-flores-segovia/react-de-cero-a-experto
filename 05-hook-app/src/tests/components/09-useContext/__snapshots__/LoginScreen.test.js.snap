// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests inside file 'LoginScreen.test.js' 'LoginScreen' component must show correctly 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="inputEmail"
      >
        Email address
      </label>
      <input
        aria-describedby="emailHelp"
        className="form-control"
        id="inputEmail"
        name="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="inputPassword"
      >
        Password
      </label>
      <input
        className="form-control"
        id="inputPassword"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <button
      className="btn btn-primary"
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
